tests/test_loop.py

```diff
@@ -43,3 +43,14 @@ def test_default_loop(self) -> None:
 
 def test_default_loop(loop: Any) -> None:
     assert asyncio.get_event_loop() is loop
+
+
+def test_setup_loop_non_main_thread() -> None:
+    def target() -> None:
+        with loop_context():
+            pass
```

Comment:
Could we run something in this loop and maybe process some signal? This is what may go wrong potentially.

Reply:
Setting up the loop context in python 3.7 still uses `SafeChildWatcher`, which in turn registers signals. I've updated the test to expect failure on python 3.7 and success in 3.8+. This way we have both code paths covered.

Subsequently after merging `pytest-xdist`, we might see occasional failures on python 3.7 tests in CI for the same reason. This could happen with any tests that use the asyncio event loop, not specific to this test alone. This will be ok once we deprecate 3.7.

Reply:
I'd rather vendor a backport and use it under Python 3.7. Also, it's not yet time to drop it anyway. aiohttp 3.8 still supports Python 3.6 even. It'll be EOL in 5 months but for CPython 3.7 there are 2 years to go.
Being a framework/library forces us to support a wider range of versions, unlike projects that are just apps bound to a single env.

Reply:
Yup, agree that backporting is better.

Reply:
here's the backport with manually resolved merge conflicts https://github.com/aio-libs/aiohttp/pull/5919